File: INTCP.m
%INTCP.m : Cp 및 Cp/T의 적분

function [dh, ds] = INTCP(T1,T2,P1,P2,A,B,C,D)
%input
% P1,P2: system 압력(bar)
% T1,T2: system 온도(K)
% A,B,C,D : Cp data
%output
% dh : Cp 의 적분 (T1-->T2)
% ds : Cp/T의 적분 (T1 -->T2)

clear out;
%기체상수
R=8.3143;    %J/mol-K

%온도차
dT=T2-T1;
sT=T2^2-T1^2;
cT=T2^3-T1^3;
qT=T2^4-T1^4;

dh=A*dT+B*sT/2.+C*cT/3.+D*qT/4.;
ds=A*log(T2/T1)+B*dT+C*sT/2.+D*cT/3.-R*log(P2/P1);

File: py/INTCP.py
import numpy as np

def INTCP(T1, T2, P1, P2, A, B, C, D):
    # 기체상수 설정
    R = 8.3143  # J/mol-K
    
    # 온도차항 계산
    dT = T2 - T1
    sT = T2**2 - T1**2
    cT = T2**3 - T1**3
    qT = T2**4 - T1**4
    
    # 엔탈피 변화 적분 (T1 -> T2)[cite: 10]
    dh = A * dT + B * sT / 2.0 + C * cT / 3.0 + D * qT / 4.0
    
    # 엔트로피 변화 적분 (T1 -> T2)[cite: 10]
    # 식: A*ln(T2/T1) + B*dT + C*sT/2 + D*cT/3 - R*ln(P2/P1)[cite: 10]
    ds = A * np.log(T2 / T1) + B * dT + C * sT / 2.0 + D * cT / 3.0 - R * np.log(P2 / P1)
    
    return dh, ds